demos/howtos/multiple_likelihoods.py

```diff
@@ -0,0 +1,180 @@
+"""
+# %% 
+# First we import the modules needed.
+import sys
+sys.path.append("../../")
+import cuqi
+import numpy as np
+import matplotlib.pyplot as plt
+from math import ceil
+
+# %%
+# Choose one of the two cases we study in this demo 
+# -------------------------------------------------
+# Choose `set_up = set_ups[0]` for the case where we have two 1D Poisson models that differ in the observation operator only. And choose `set_up = set_ups[1]` for the case where we have two 1D Poisson models that differ in the source term only.
+
+set_ups = ["multi_observation", "multi_source"]
+set_up = set_ups[0]
+
+assert set_up == "multi_observation" or set_up == "multi_source", "set_up must be either 'multi_observation' or 'multi_source'"
+
+# %%
+# Set up the parameters used in both models
+# -----------------------------------------
+
+dim = 50  # Number of the model grid points
+endpoint = 1  # The model domain is the interval [0, endpoint]
+field_type = "Step"  # The conductivity (or diffusivity) field type
+SNR = 500  # Signal-to-noise ratio
+n_steps = 2  # Number of steps in the conductivity (or diffusivity) step field
+
+# Exact solution
+x_exact = np.empty(dim)
+x_exact[:ceil(dim/2)] = 1
+x_exact[ceil(dim/2):] = 3
+
+
+# %%
+# Set up the first model
+# ----------------------
+
+observation_grid_map1 = None
+if set_up == "multi_observation":
+	# Observe on the first half of the domain
+	observation_grid_map1 = lambda x: x[np.where(x<.5)] 
+
+# The source term signal
+source1 = lambda xs: 20*np.sin(xs)+20.1
+
+# Obtain the forward model from the test problem
+model1, data1, problemInfo1 = cuqi.testproblem.Poisson_1D.get_components(dim=dim,
+                                                                         endpoint=endpoint,
+                                                                         field_type=field_type,
+                                                                         field_params={
+                                                                             'n_steps': n_steps},
+                                                                         observation_grid_map=observation_grid_map1,
+                                                                         exactSolution=x_exact,
+                                                                         SNR=SNR)
+
+# Plot data, exact data and exact solution
+plt.figure()
+data1.plot(label='data')
+problemInfo1.exactData.plot(label='exact data')
+problemInfo1.exactSolution.plot(label='exact solution')
+plt.legend()
+
+# %%
+# Set up the second model
+# -----------------------
+
+observation_grid_map2 = None
+if set_up == "multi_observation":
+        # Observe on the second half of the domain
+	observation_grid_map2 = lambda x: x[np.where(x>=.5)]
+
+# The source term signal
+if set_up == "multi_source":
+	source2 = lambda xs: 20*np.sin(2*xs)+20.1
```

Comment:
It took me a little while to determine what the difference between the two likelihoods were in the multi source case, as the code reads almost the same, only a "2*" hidden inside the sine. The plots of the data for the two different sources look almost identical. Could we make the difference clearer somehow? For example could we make plots of the two source terms?

Reply:
This was a bug because I did not pass the source term to the testproblem. Now it is fixed :)